server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -547,6 +548,10 @@ private int submitCompactionTasks(
         slotsRequiredForCurrentTask = findMaxNumTaskSlotsUsedByOneNativeCompactionTask(config.getTuningConfig());
       }
 
+      if (entry.getCurrentStatus() != null) {
```

Comment:
Nice!

To be on the safe side, could you please verify if this task context gets written out in the final task payload and also the task report?

Reply:
yes verified.